Made validation errors better to work with 400 exceptions.

boralyl opened the pull request:
- returns all missing required fields in the error message
- Specifies which parameter the error corresponds to
- returns all errors for all params in the exception's errors attr

Commit:
Made validation errors better to work iwth in 400 exceptions.

- returns all missing required fields in the error message
- Specifies which parameter the error corresponds to
- returns all errors for all params in the exception's errobj attr

Commit:
Releasing v3.0.0-b.4

noonat commented on doctor/errors.py:
```diff
@@ -10,17 +10,22 @@ class TypeSystemError(Exception):
     :param detail: Detail about the error.
     :param cls: The class type that was being instantiated.
     :param code: The error code.
+    :param errobj: A dict containing all validation errors during the request.
```
I've never been a huge fan of this name from our internal usage. Maybe it would be better to name it something else for this public version? validation_errors? param_errors? details extra? Not sure...

noonat commented on docs/_logic_functions.rst:
```diff
@@ -18,6 +18,11 @@ Any argument without a default value is considered required while others are
 
+Any parameters that don't validate will raise a :class:`~doctor.flask.HTTP400Exception`.
+This exception will contain all validation errors and missing required properties.
+If there are more than one error, you can access them from the exception's
```
Minor, typo: If there is more than one error

noonat commented on doctor/flask.py:
```diff
@@ -49,6 +53,12 @@ def __str__(self):
 
 
 class HTTP400Exception(SchematicHTTPException, BadRequest):
+    """Reperesents a HTTP 400 error.
```
Minor, typo: Represents

boralyl committed:
Fixed some typos and renamed errobj to errors.

boralyl committed:
Removed the param name prefix from the error message.

It only gets added if there is a single error.  Otherwise the user can
get the param/error combo from the errors attribute.

boralyl committed:
Fixed error message
